datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -2309,14 +2309,12 @@ mod test {
         round_trip_type(DataType::Decimal128(10, 2))?;
         round_trip_type(DataType::Decimal256(30, 2))?;
 
-        for nullable in [true, false] {
-            round_trip_type(DataType::List(
-                Field::new_list_field(DataType::Int32, nullable).into(),
-            ))?;
-            round_trip_type(DataType::LargeList(
-                Field::new_list_field(DataType::Int32, nullable).into(),
-            ))?;
-        }
+        round_trip_type(DataType::List(
```

Comment:
Should we also add a test here showing that the List becomes nullable after roundtrip? That might additionally document that this is an intended behavior rather than a bug. 

I do see you have added a comment which I think is probably enought

Reply:
I don't think we need to test the null coercion specifically, since that's not desired behavior necessarily. But I added a test case to confirm that we can now read plans with non-nullable lists: https://github.com/apache/datafusion/pull/10874/commits/a25abd89a5547182193caee68a7c6f458f4b1d23